compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp

```diff
@@ -0,0 +1,215 @@
+  const_node->rank(1);
+  const_node->dim(0) = values.size();
+
+  for (int32_t i = 0; i < values.size(); i++)
+  {
+    const_node->at<loco::DataType::S32>(i) = values.at(i);
+  }
+
+  return const_node;
+}
+
+/**
+ * @brief Return newly-created CircleReshape node
+ */
+luci::CircleNode *build_reshape(loco::Graph *graph, const std::string &name,
+                                const std::shared_ptr<luci::CircleNodeOrigin> &origin,
+                                luci::CircleNode *input, const std::vector<uint32_t> new_shape)
+{
+  auto reshape_node = graph->nodes()->create<luci::CircleReshape>();
+  reshape_node->tensor(input);
+  reshape_node->name(name);
+  luci::add_origin(reshape_node, origin);
+
+  auto new_shape_const = build_rank1_const(graph, new_shape);
+  {
+    new_shape_const->name(name + "/new_shape");
+    luci::add_origin(new_shape_const, origin);
+  }
+
+  reshape_node->shape(new_shape_const);
+
+  return reshape_node;
+}
+
+/**
+ * @brief Return value in position on CircleConst with int64 format.
+ */
+int64_t value_from_circle_const(const luci::CircleConst *node, uint32_t idx)
+{
+  assert(node->rank() == 1 && node->dim(0).value() > idx);
+  assert(node->dtype() == loco::DataType::S64 || node->dtype() == loco::DataType::S32);
+
+  if (node->dtype() == loco::DataType::S64)
+    return node->at<loco::DataType::S64>(idx);
+  return static_cast<int64_t>(node->at<loco::DataType::S32>(idx));
+}
+
+bool substitute_strided_slice_to_reshape(luci::CircleStridedSlice *ss_node)
+{
+  if (ss_node->shrink_axis_mask() == 0)
+    return false;
+
+  // TODO Consider cases with ellipsis_mask and new_axis_mask
+  // NOT YET SUPPORTED
+  if (ss_node->ellipsis_mask() != 0 or ss_node->new_axis_mask() != 0)
+    return false;
+
+  auto begin_const = dynamic_cast<luci::CircleConst *>(ss_node->begin());
+  auto strides_const = dynamic_cast<luci::CircleConst *>(ss_node->strides());
+  auto end_const = dynamic_cast<luci::CircleConst *>(ss_node->end());
+  auto input_node = loco::must_cast<luci::CircleNode *>(ss_node->input());
+
+  if (not(begin_const && strides_const && end_const && input_node))
+    return false;
+
+  // condition check
+  std::bitset<32> begin_mask(ss_node->begin_mask());
+  std::bitset<32> end_mask(ss_node->end_mask());
+  std::bitset<32> shrink_axis_mask(ss_node->shrink_axis_mask());
+
+  uint input_rank = input_node->rank();
+  for (uint32_t i = 0; i < input_rank; i++)
+  {
+    if (!input_node->dim(i).known())
+      return false;
+
+    auto begin_dim = value_from_circle_const(begin_const, i);
+    if (not(begin_dim == 0 or begin_mask.test(i) == true))
+      return false;
+
+    // NOTE:
+    //    In Tensorflow and TFLite, e.g., if input_shape = [2,3],
+    //    strided_slice.end = [10,20] (larger value than actual dim)
+    //    is treated as strided_slice.end = [2,3]
+    int64_t end_dim = value_from_circle_const(end_const, i);
+    if (not(end_dim >= input_node->dim(i).value() or end_mask.test(i) == true))
```

Comment:
(optional) ditto.
```suggestion
    if (end_dim < input_node->dim(i).value() and end_mask.test(i) != true))
```
Is this intended? I suggested because of readability.

Reply:
I my case, the way I wrote was easy to understand. 
But I don't have strong taste. I will revise as you suggested.